src/android/GeofenceTransitionsIntentService.java

```diff
@@ -0,0 +1,256 @@
+import android.support.v4.app.TaskStackBuilder;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.google.android.gms.location.Geofence;
+import com.google.android.gms.location.GeofencingEvent;
+
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Listener for geofence transition changes.
+ *
+ * Receives geofence transition events from Location Services in the form of an Intent containing
+ * the transition type and geofence id(s) that triggered the transition. Creates a notification
+ * as the output.
+ */
+public class GeofenceTransitionsIntentService extends IntentService{
+
+    protected static final String TAG = "GeofenceTransitionsIS";
+    private double lat, lon, alt;
+    private String timestamp = "yyyyMMdd'T'HH:mm:ss";
+
+    public DBHelper GPS_dbManager;
+    public Geofence geofence;
+    String provider;
+    Location mLastLocation;
+    LocationManager mLocationManager = null;
+
+    private static final int LOCATION_INTERVAL = 1000;
+    private static final float LOCATION_DISTANCE = 10f;
+
+
+
+    /**
+     * This constructor is required, and calls the super IntentService(String)
+     * constructor with the name for a worker thread.
+     */
+    public GeofenceTransitionsIntentService() {
+        // Use the TAG to name the worker thread.
+        super(TAG);
+    }
+
+    private class GetLocationInfo implements LocationListener{
+
+        public GetLocationInfo(String provider1){
+            mLastLocation = new Location(provider1);
+        }
+        @Override
+        public void onLocationChanged(Location location) {
+            lat = location.getLatitude();
+            lon = location.getLongitude();
+            alt = location.getAltitude();
+
+            SimpleDateFormat GPStime = new SimpleDateFormat("yyyyMMdd'T'HH:mm:ss");
+            timestamp = GPStime.format (location.getTime());
+        }
+
+        @Override
+        public void onStatusChanged(String s, int i, Bundle bundle) {}
+
+        @Override
+        public void onProviderEnabled(String s) {}
+
+        @Override
+        public void onProviderDisabled(String s) {}
+    }
+
+    GetLocationInfo[] mLocationListeners = new GetLocationInfo[]{
+            new GetLocationInfo(LocationManager.NETWORK_PROVIDER)
+    };
+
+    private void initializeLocationManager() {
+        Log.e(TAG, "initializeLocationManager");
+        if (mLocationManager == null) {
+            mLocationManager = (LocationManager) getApplicationContext()
+                    .getSystemService(Context.LOCATION_SERVICE);
+        }
+    }
+    /**
+     * Handles incoming intents.
+     * @param intent sent by Location Services. This Intent is provided to Location
+     *               Services (inside a PendingIntent) when addGeofences() is called.
+     */
+    @Override
+    protected void onHandleIntent(Intent intent) {
+
+        initializeLocationManager();
```

Comment:
here's a sequence diagram that shows why this won't work

![image](https://cloud.githubusercontent.com/assets/859611/13216508/755e623e-d95c-11e5-856a-a0b59b729d23.png)